bundles/org.openhab.binding.jeelink/src/main/java/org/openhab/binding/jeelink/internal/SensorDefinition.java

```diff
@@ -96,6 +102,18 @@ public static ThingHandler createHandler(ThingTypeUID thingTypeUid, Thing thing)
         return null;
     }
 
+    public static List<JeeLinkReadingConverter<?>> getDiscoveryConverters() {
+        synchronized (CONVERTERS) {
```

Comment:
If you make it a `Set` and initialize it as `ConcurrentHashMap.newKeySet()` you can omit the synchronized.

Reply:
How is that supposed to work? There has to be some magic going on if the map can correctly synchronize this by itself. There are several calls to the map in the synchronized block.

Reply:
You can directly initialize it in the declaration, remove the check here and the static block above.

```
    private static final Set<SensorDefinition<?>> SENSOR_DEFS = Stream
            .of(new LaCrosseSensorDefinition(), new Ec3kSensorDefinition(), new Pca301SensorDefinition(),
                    new Tx22SensorDefinition(), new RevoltSensorDefinition()).collect(Collectors.toSet());
    private static final Set<JeeLinkReadingConverter<?>> CONVERTERS = SENSOR_DEFS.stream()
            .map(SensorDefinition::createConverter).collect(Collectors.toSet());
```

Reply:
Yes, I can. But i'd prefer not to, for readability reasons.

Reply:
Your code is much more complex and difficult (because it splits the initialization in three parts: initialize the sets in the declaration, add the sensor defs in the `static` block and add the converters when they are first requested).